src/components/widgets/Form.js

```diff
@@ -0,0 +1,315 @@
+import React, { Component } from 'react'
+import { RadioGroup, Radio } from 'react-radio-group'
+import Button from '../Button.js'
+import "../../styles/form.css"
+
+export default class Form extends Component {
+  constructor( props ) {
+    super( props )
+    this.state = {}
+    this.setFieldState = this.setFieldState.bind( this )
+  }
+
+  componentWillMount() {
+    this.setupState()
+  }
+
+  setupState() {
+    const { sections } = this.props
+    sections.forEach( section => {
+      section.fields.forEach ( field => {
+        const { name, defaultValue } = field
+        this.setState({[ name ]: defaultValue })
+        if ( field.type === "nested-dropdown" ) {
+          const nestedDropdownValue = field.options.filter( option => {
+            return option.value === defaultValue
+          })[0].field
+          this.setState({
+            [ nestedDropdownValue.name ]: nestedDropdownValue.defaultValue
+          })
+        }
+      })
+    })
+  }
+
+  setFieldState( state, stateValue ) {
+    this.setState({
+      [ state ]: stateValue
+    })
+  }
+
+  submitWidget( state ) {
+    const widgetData = {}
+    for ( let key in state ) {
+      widgetData[key] = state[key]
+      widgetData.y = '10px'
```

Comment:
We were thinking last week that we didn't know where they were supposed to render. On geckoboard they render next to any other widgets on the dashboard. Should we implement that feature now? In the meantime, we just hardcoded these in to give the front end widgets team this data that they were looking for.